### PR TITLE
Add benchmarks

### PR DESCRIPTION
**Pull request by SergeyMokin**

### CODE REVIEW

**Comment by SergeyMokin** [comment]
Will fix conflicts tomorrow :)

File: perf/ListPool.Benchmarks/ListPoolContainsBenchmark.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        public void IterationCleanup()
+        {
+            _listPool.Dispose();
+        }
+
+        [Benchmark(Baseline = true)]
+        public void List()
+        {
+            _list.Contains(1);
+            _list.Contains(N / 2);
```
Comment (faustodavid) [inline perf/ListPool.Benchmarks/ListPoolContainsBenchmark.cs]: I think just picking one should give us good stats.

File: perf/ListPool.Benchmarks/ListPoolIndexOfBenchmarks.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        public void IterationCleanup()
+        {
+            _listPool.Dispose();
+        }
+
+        [Benchmark(Baseline = true)]
+        public void List()
+        {
+            _list.IndexOf(1);
+            _list.IndexOf(N / 2);
```
Comment (faustodavid) [inline perf/ListPool.Benchmarks/ListPoolIndexOfBenchmarks.cs]: Just picking one IndexOf should be good enough

File: perf/ListPool.Benchmarks/ListPoolContainsBenchmark.cs (hunk 1 above)
Comment (SergeyMokin) [reply perf/ListPool.Benchmarks/ListPoolContainsBenchmark.cs]: done :)

File: perf/ListPool.Benchmarks/ListPoolIndexOfBenchmarks.cs (hunk 2 above)
Comment (SergeyMokin) [reply perf/ListPool.Benchmarks/ListPoolIndexOfBenchmarks.cs]: done :)

**Comment by SergeyMokin** [comment]
conflicts again :D

File: tests/ListPool.UnitTests/ListPoolTests.cs [hunk 3]
```diff
@@ -13,9 +14,10 @@ public class ListPoolTests
         public void Add_item_without_indicate_capacity_of_list()
         {
             int expectedItem = s_fixture.Create<int>();
-            using var sut = new ListPool<int>();
-
-            sut.Add(expectedItem);
+            using var sut = new ListPool<int>
```
Comment (faustodavid) [inline tests/ListPool.UnitTests/ListPoolTests.cs]: Why we should specify the scope of the using?

File: tests/ListPool.UnitTests/ListPoolExtensionsTests.cs [hunk 4]
```diff
@@ -33,7 +33,7 @@ public void ToListPool_when_source_is_null_throw_ArgumentNullException()
             IEnumerable<int> source = null;
             string expectedName = nameof(source);
 
-            var exception = Assert.Throws<ArgumentNullException>(() => source.ToListPool());
+            ArgumentNullException exception = Assert.Throws<ArgumentNullException>(() => source.ToListPool());
```
Comment (faustodavid) [inline tests/ListPool.UnitTests/ListPoolExtensionsTests.cs]: Why not use **var** here?

Comment (SergeyMokin) [reply tests/ListPool.UnitTests/ListPoolExtensionsTests.cs]: recommendation after merge this: https://github.com/faustodavid/ListPool/blob/master/.editorconfig

File: tests/ListPool.UnitTests/ListPoolTests.cs (hunk 3 above)
Comment (SergeyMokin) [reply tests/ListPool.UnitTests/ListPoolTests.cs]: just editor recommendation again, I can move it back :D

**Comment by SergeyMokin** [comment]
About fail you already resolved it in your PR, I will merge your changes when it will be in master

**Comment by faustodavid** [comment]
> About fail you already resolved it in your PR, I will merge your changes when it will be in master

done :)

**Comment by SergeyMokin** [comment]
build fine now

**Pull request merged**